roles/reproducer/tasks/validations.yml

```diff
@@ -83,3 +83,58 @@
       cifmw_network_dnsmasq_config is set and not empty. This feature is NOT
       supported anymore. Please contact the CI Framework team to check your
       configuration, and see if you really need to pass anything to dnsmasq.
+
+- name: Verify cifmw_arch_automation_file is undefined
+  ansible.builtin.assert:
+    quiet: true
+    that:
+      - cifmw_arch_automation_file is undefined
+    msg: >-
+      cifmw_arch_automation_file is defined. This parameter is NOT supported
+      anymore. You can use cifmw_architecture_automation_file instead, providing
+      the absolute path to the automation file. Note that you shouldn't need to
+      set it at all.
+
+# Ensure we don't mix upstream and midstream content
+- name: Ensure we're consuming consistent project sources
+  when:
+    - cifmw_reproducer_validate_project_sources | default(true) | bool
+  vars:
+    _custom_rep_dict: >-
+      {{
+        cifmw_reproducer_repositories |
+        items2dict(key_name='dest', value_name='src')
+      }}
+    # Don't consider local paths, only remotes.
+    # This should prevent failing whenver we sync local repositories
+    # while developping with the Framework.
+    _projects_src: >-
+      {{
+        cifmw_reproducer_default_repositories |
```

Comment:
shouldn't we worry about default values for this var here? IIRC it defaults to upstream

Reply:
we need to extend the whole thing. For now, it doesn't check the value, but only the consistency. It's weak, to be fair, and we probably want/need something better here, to ensure we're in the right context.

I don't know if we could get anything from within zuul env showing "where" we run, and create some "validation map" stating something like this:
```yaml
cifmw_src_base_validation:
  upstream:
    src_base: https://github.com/openstack-k8s-operators
    branch: main
  downstream_trunk:
    src_base: /home/zuul/src/....
    branch: trunk  # whatever it should be
  downstream_18_proposed:
    src_base: https://github.com/openstack-k8s-operators.
    branch: 18.0-proposed  # whatever it should be
```

That would really consolidate the whole thing. Any thoughts? Of course, it must be extensible in an easy way, to be future proof (new branch, new entry in the mapping).